src/App.js

```diff
@@ -14,11 +16,111 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+let productList = [];
+
+class App extends React.Component {
+  state = {
+    product: ' none',
+  };
+
+  selectProduct = (event) => {
+    const deleteButton = document.querySelector('.delete');
+
+    deleteButton.style.display = 'block';
+    const selectedProductText = event.target.parentElement
+      .textContent.replace('AddRemove', '');
+
+    productList.push(selectedProductText);
+
+    if (event.target.tagName === 'BUTTON') {
+      event.target.parentElement.classList.add('selected');
+      event.target.style.display = 'none';
+      event.target.nextSibling.style.display = 'block';
+    }
+
+    this.setState({ product: productList.join(', ') });
+  };
+
+  deleteProduct = () => {
+    const selectedProduct = [...document.querySelectorAll('.selected')];
+    const removeButtons = [...document.querySelectorAll('.removeButton')];
+    const addButtons = [...document.querySelectorAll('.addButton')];
+    const deleteButton = document.querySelector('.delete');
+
+    removeButtons.forEach(removeButton => removeButton.style.display = 'none');
+    addButtons.forEach(addButton => addButton.style.display = 'block');
+
+    productList = [];
+
+    selectedProduct.forEach(product => product.classList.remove('selected'));
+    deleteButton.style.display = 'none';
+    this.setState({ product: ' none' });
+  }
+
+  removeProduct = (event) => {
+    event.target.parentElement.classList.remove('selected');
+    event.target.style.display = 'none';
+    event.target.previousSibling.style.display = 'block';
+    const index = productList.indexOf(event.target.parentElement
+      .textContent.replace('AddRemove', ''));
+
+    const deleteButton = document.querySelector('.delete');
+
+    if (index > -1) {
+      productList.splice(index, 1);
+    }
+
+    this.setState({ product: productList.join(', ') });
+
+    if (productList.length === 0) {
+      deleteButton.style.display = 'none';
+      this.setState({ product: ' none' });
+    }
+  }
+
+  render() {
+    const { product } = this.state;
+
+    return (
+      <div className="App">
+        <div className="header">
+          <h1>
+            Selected good: -
+            {' '}
+            {product}
+          </h1>
+          <button
+            className="delete"
+            type="button"
+            onClick={this.deleteProduct}
+          >
+            X
+          </button>
+        </div>
+        <ul>
+          {goodsFromServer.map(good => (
+            <li key={good}>
+              {good}
+              <button
+                className="addButton"
+                type="submit"
+                onClick={this.selectProduct}
+              >
+                Add
+              </button>
+              <button
+                className="removeButton"
+                type="button"
+                onClick={this.removeProduct}
```

Comment:
```suggestion
                onClick={() => this.removeProduct(good)}
```
Передавай в функцию good, таким образом ты будешь знать какой именно продукт нужно удалить/добавить